src/ScalarPotentials/ScalarPotential.jl

```diff
@@ -48,14 +48,35 @@ function get_2π_potential(sp::ScalarPotential{T, 3, Cylindrical}, axφ::Discret
     for idx_n in 1:n
         for il in 1:l
             idx::Int = il + (idx_n - 1) * l
-            new_ticks[idx] = (idx_n - 1) * Δφ + axφ.ticks[il]
+            new_ticks[idx] = mod((idx_n - 1) * Δφ + axφ.ticks[il], T(2π))
             new_pot[:, idx, :] = sp[:, il, :]
         end
     end
-    new_axφ = DiscreteAxis{AT, :periodic, :periodic}( new_int, new_ticks )
-    new_axes = (sp.grid[1], new_axφ, sp.grid[3])
-    new_grid = Grid{AT, 3, Cylindrical, typeof(new_axes)}( new_axes )
-    ScalarPotential(sp, new_pot, new_grid)
+    P = sortperm(new_ticks)
+    if new_ticks[P][1] == 0
+        new_axφ = DiscreteAxis{AT, :periodic, :periodic}( new_int, new_ticks[P] )
+        new_axes = (sp.grid[1], new_axφ, sp.grid[3])
+        new_grid = Grid{AT, 3, Cylindrical, typeof(new_axes)}( new_axes )
+        ScalarPotential(sp, new_pot[:,P,:], new_grid)
+    else
+        new_ticks_new_zero::Vector{AT} = Vector{AT}(undef, l * n + 1)
+        new_ticks_new_zero[1] = T(0)
+        new_ticks_new_zero[2:end] = new_ticks[P]
+        new_pot_new_zero = Array{eltype(sp.data), 3}(undef, size(sp, 1), l * n + 1, size(sp, 3))
+        new_pot_new_zero[:,1,:] = (eltype(sp.data) <: AbstractFloat) ? sp[:,1,:] + 
+                        (sp[:,end,:] - sp[:,1,:]) * (T(0) - new_ticks_new_zero[2]) / (T(2π) - T(new_ticks_new_zero[end]) * n + T(new_ticks_new_zero[2])) :
```

Comment:
I feel like this formula is wrong and it should contain something like `Δφ * n` rather than `T(new_ticks_new_zero[end]) * n`.. I will do the calculation and see if this needs to be adjusted.

Reply:
The permutation of the new potential based on the permutation of the new ticks is missing here, I will push some changes to fix this..